crates/accelerate/src/synthesis/permutation/mod.rs

```diff
@@ -82,11 +82,38 @@ fn _synth_permutation_acg(py: Python, pattern: PyArrayLike1<i64>) -> PyResult<Ci
     )
 }
 
+/// Synthesize a permutation circuit for a linear nearest-neighbor
+/// architecture using the Kutin, Moulton, Smithline method.
+#[pyfunction]
+#[pyo3(signature = (pattern))]
+pub fn _synth_permutation_depth_lnn_kms(py: Python, pattern: PyArrayLike1<i64>) -> PyResult<CircuitData> {
+    let inverted = utils::invert(&pattern.as_array());
+    let view = inverted.view();
+    let num_qubits = view.len();
+
+    let swap_layer: Vec<(usize, usize)> = utils::_create_swap_layer(&view);
+        
```

Comment:
The original Python code called this function multiple times with alternating odd/even layers:
```
for i in range(num_qubits):
    _create_swap_layer(qc, cur_pattern, i % 2)
```
And here it seems that you are only calling the function once.